### PR TITLE
Update docs for 2.0.0-M2 and 2.0.0-M3

### PR DESCRIPTION
**Pull request by travisbrown**
The usual. That's a bunch of new contributors.

r? @rossabaker

### CODE REVIEW

**Comment by travisbrown** [comment]
@kailuowang What do you mean? Do we not include doc / README editors here? I just ran `git log v2.0.0-M1...v2.0.0-M3` and added these manually from there.

**Comment by kailuowang** [comment]
I was referring to the authors mentioned in changes

**Comment by kailuowang** [comment]
@travisbrown sorry I accidently just updated the PR directly, I meant to create a PR to your branch but github throws me a 404 when I tried. Then I clicked the wrong button... 
For the most part I used [this script](https://github.com/typelevel/cats/blob/master/scripts/releaseNotes.scala) which requires some work on github beforehand, i.e. setting milestone and labels for all merged PRs. 
I removed trivial README updates from the release notes as well as corrected some author names.  authors given by`git logs` are not accurate for squash merged PRs.

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,79 @@
...
+* [#2821](https://github.com/typelevel/cats/pull/2821) removing unused resolver  by @kailuowang
+        
+
+### 2 test improvments
+
+* [#2831](https://github.com/typelevel/cats/pull/2831) fixed applicative monoid test  by @kailuowang
+* [#2827](https://github.com/typelevel/cats/pull/2827) In Apply.semigroup test replace ` ListWrapper` with `Option`  by @kailuowang
+        
+
+### 1 bug fixes
```
Comment (travisbrown) [inline CHANGES.md]: I know this is automated but can we at least fix the pluralization?

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,79 @@
...
+### 5 build improvements
+
+* [#2858](https://github.com/typelevel/cats/pull/2858) Don't create the hydra.license if $HydraLicense is empty  by @dotta
+* [#2857](https://github.com/typelevel/cats/pull/2857) Updates for 2.13.0-RC2  by @travisbrown
+* [#2856](https://github.com/typelevel/cats/pull/2856) Update sbt-scalajs to 0.6.28  by @scala-steward
+* [#2848](https://github.com/typelevel/cats/pull/2848) Enables Hydra on cats build  by @kailuowang
+* [#2821](https://github.com/typelevel/cats/pull/2821) removing unused resolver  by @kailuowang
+        
+
+### 2 test improvments
```
Comment (travisbrown) [inline CHANGES.md]: And the spelling.

Comment (LukaJCB) [inline CHANGES.md]:
```suggestion
### 2 test improvements
```

File: CHANGES.md (hunk 1 above)
Comment (LukaJCB) [inline CHANGES.md]:
```suggestion
### 1 bug fix
```

File: CHANGES.md [hunk 3]
```diff
@@ -64,13 +64,13 @@ Cats 2.0.0-M2 breaks binary compatibility on cats-laws, cats-kernel-laws, cats-t
...
 
-### 2 test improvments
+### 2 test improvements
 
 * [#2831](https://github.com/typelevel/cats/pull/2831) fixed applicative monoid test  by @kailuowang
 * [#2827](https://github.com/typelevel/cats/pull/2827) In Apply.semigroup test replace ` ListWrapper` with `Option`  by @kailuowang
         
 
-### 1 bug fixes
+### 1 bug fixe
```
Comment (LukaJCB) [inline CHANGES.md]: close, but not quite 😝

**Comment by kailuowang** [comment]
sort of related. https://github.com/typelevel/cats/pull/2881 is submitted to update the Process document.

**Comment by kailuowang** [comment]
@LukaJCB will you have time to take another quick look?

**Pull request merged**